omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -112,33 +74,30 @@ exports.login = function (req, res) {
  * 
  * @param {*} req contains request headers and payload businessEmail,password
  */
-exports.getToken = function (req, res) {
-    const body = req.body;
-    if (body && body.provider === 'GOOGLE') {
-        User.findUser(body.email, function (err, user) {
+exports.getToken = async function (req, res) {
+    try {
+        const body = req.body;
+        if (body && body.provider === 'GOOGLE') {
+            const [user] = await User.findUser(body.email)
             if (user && user.userId) {
-                    delete user.password;
-                    const token = jwt.sign(user, process.env.JWT_SECRET, {
-                        expiresIn: process.env.EXPIRE_IN
-                    });
-                    res.status(200).json({
-                        success: true,
-                        message: 'Logged in successfully',
-                        userInfo: user,
-                        authToken: token
-                    });
-                
-            } else
-                res.status(401).json({
-                    success: false,
-                    message: 'please register your account',
-                    redirect: 'sign-up'
+                delete user.password;
+                if (user.provider === '' || user.provider === null) {
+                    user.provider = body.provider;
+                    user.photoUrl = body.photoUrl;
+                    await User.update(user)
+                }else if(user.provider !== 'GOOGLE'){
+                    return res.status(422).json({ success: false, message: 'The user is not a Google user.' });
```

Comment:
I'm not sure exactly how easy it is to access this function from the outside, but this error also returns an indication that the user actually exists.

Reply:
the flow of the function is changed, as it will always be a issue.
the new flow is 
1. user can sign-in with google.
2. checks Google-User email-id already registered with omegatrade db or not.
3. if already registered - update photoUrl to registered user.
4. else create a new user with random password(password should not be empty) and force user to change password if onboarded for the first time.

note : the reason for step 4 -  As we are not using mailing service to share end user with password or link, we are forcing user to change password in first time login event.